client/src/extension.ts

```diff
@@ -235,21 +233,14 @@ export function activate(context: ExtensionContext) {
   // Start the client. This will also launch the server
   client.start();
 
-  // Restart the language client automatically when certain configuration
-  // changes. These are typically settings that affect the capabilities of the
-  // language client, and because of that requires a full restart.
   context.subscriptions.push(
     workspace.onDidChangeConfiguration(({ affectsConfiguration }) => {
-      // Put any configuration that, when changed, requires a full restart of
-      // the server here. That will typically be any configuration that affects
-      // the capabilities declared by the server, since those cannot be updated
-      // on the fly, and require a full restart with new capabilities set when
-      // initializing.
       if (
-        affectsConfiguration("rescript.settings.inlayHints") ||
-        affectsConfiguration("rescript.settings.codeLens")
+        affectsConfiguration("rescript.settings")
       ) {
-        commands.executeCommand("rescript-vscode.restart_language_server");
+        client.sendNotification(
+          "workspace/didChangeConfiguration",
+          { settings: workspace.getConfiguration("rescript.settings") }).catch((err) => { window.showErrorMessage(String(err)) })

```

Comment:
While I do like this touch of notifying the server on config changes, this in combination with removing the pulling breaks all other clients than VSCode, unless they _also_ implement this pushing. I'd rather pull and just have it work on all clients without the extra work needed.